File: 1995/2-4.py
#!/usr/bin/python
ary = ["0","0","0","0","0","0","0"]
chk = ["1","0","0","0","0","0","1"]
fin = open("prob4.in", "r")
q = []
p = []
while 1:
	line = fin.readline()
	if line == "":
		break
	if line[-1] == "\n":
		line = line[:-1]
	for i in range(len(line)):
		q.append(line[i])
while q != []:
	for i in range(0,7):
		ary[i] = q.pop(0)
	if ary == chk:
		while 1:
			cnt = 0
			for i in range(0,7):
				ary[i] = q.pop(0)
			if ary == chk:
				break
			for i in range(0,7):
				cnt += int(ary[i]) * 2 ** (6 - i)
			p.append(chr(cnt))
print "".join([i for i in p])

File: 1995/4/2-4.cpp
#include <iostream>
#include <string>
#include <cmath>
using namespace std;

int main()
{
    string ary = "0000000", chk = "1000001";
    int cnt;
    for (int i = 0; i < 6; i++)
        cin >> ary[i];
    while (cin >> ary[6])
    {
        if (ary == chk)
        {
            while (true)
            {
                cnt = 0;
                for (int i = 0; i < 7; i++)
                    cin >> ary[i];
                if (ary == chk) break;
                for (i = 0; i < 7; i++)
                    cnt += (int)(ary[i] - '0') * pow(2, 6 - i);
                cout << (char)cnt;
            }
        }
        for (int i = 0; i < 6; i++)
            ary[i] = ary[i + 1];
    }
    cout << endl;
}
